src/browser/main.js

```diff
@@ -2742,10 +2742,11 @@ function init_ui(profile, settings, emulator)
             {
                 const phone_keyboard = document.getElementsByClassName("phone_keyboard")[0];
 
-                // stop mobile browser from scrolling into view when the keyboard is shown
-                phone_keyboard.style.top = document.body.scrollTop + 100 + "px";
-                phone_keyboard.style.left = document.body.scrollLeft + 100 + "px";
+                phone_keyboard.style.top = window.scrollY + e.clientY + 20 + "px";
+                phone_keyboard.style.left = window.scrollX + e.clientX + "px";
 
+                // clean after previous input
+                phone_keyboard.value = "";
                 phone_keyboard.focus();
             }
```

Comment:
The removal of the `mousedown` means that this line is only hit if they mouse is disabled. Latest buildroot enables the mouse, so clicking on the screen does not pop up the mobile keyboard any more.

I'd suggest moving this block (including the `if(window.getSelection().isCollapsed)`) out of the else branch

Reply:
I thought the keyboard shouldn't be displayed when the mouse in the guest OS is on, because the virtual keyboard takes up a lot of space on the phone screen.